src/pynwb/base.py

```diff
@@ -497,6 +497,26 @@ def data(self):
         # load the data from the timeseries
         return self.timeseries.data[self.idx_start: (self.idx_start + self.count)]
 
+    @classmethod
+    @docval({'name': 'timeseries', 'type': TimeSeries, 'doc': 'the timeseries object to reference.'})
+    def empty(cls, timeseries):
+        """
+        Creates an empty TimeSeriesReference class to represent missing data.
+
+        When missing data needs to be represented, NWB defines ``None`` for the complex data type ``(idx_start,
+        count, TimeSeries)`` as (-1, -1, TimeSeries) for storage. The exact timeseries object will technically not
+        matter since the empty reference is a way of indicating a NaN value in a
+        :py:class:`~pynwb.base.TimeSeriesReferenceVectorData` column.
+
+        An example where this functionality is used is :py:class:`~pynwb.icephys.IntracellularRecordingsTable`
+        where only one of stimulus or response data was recorded. In such cases, the timeseries object for the
+        empty stimulus :py:class:`~pynwb.base.TimeSeriesReference` could be set to the response series, or vice versa.
```

Comment:
This seems like an inconvenience and I think it would confuse users. What about creating a dummy TimeSeries to use here? Or using None as the object reference?

Reply:
The problem with creating a dummy TimeSeries is that it has to be added somewhere...

Reply:
@rly I agree it's confusing . We discussed this a bit above


> Good question. I think having an empty `TimeSeries` to point to makes sense. I'm not sure what the best way is to autogenerate that, since we would need to also add it to the file. For now, maybe we can still require that a user to provide the `TimeSeries` and maybe have a best practice on how do to this. Let's discuss when we meet next time how we may be able to do this without the user providing a `TimeSeries`. The `TimeSeries` is necessary because this is stored as compound data type, so all elements must be set, and we don't have a NaN value for ints and object references, so `(-1, -1, <some ref>)` is just a way of saying NaN or invalid. Technically, it really doesn't matter where the reference points to, so maybe there is a way that we can avoid needed a `TimeSeries` at all.

Reply:
right now `IntracellularRecordingsTable` automatically [fills in the missing stimulus / response](https://github.com/NeurodataWithoutBorders/pynwb/blob/dev/src/pynwb/icephys.py#L575)  - so having a dummy TimeSeries might be helpful there as well?

Reply:
@oruebel and I discussed this briefly and decided to leave this PR be and ask @stephprince to try to set the object reference to None in a separate PR.